Display X

SimonRhook opened the pull request:
# Overview

This usage example demonstrates a way to use the display_x function to retrieve the x coordinates for a given display

# Files Included

Splashkit Function: `display_x`

# Files Included
- [x] Title and explanation (.txt)
- [x] C++ code (SplashKit)
- [] C++ code (Beyond SplashKit)
- [x] C# code (top-level statements)
- [x] C# code (Object-Oriented Programming)
- [x] Python code
- [x] Screenshot

# Usage Example Checks
- [x] Simple, clear demonstration of the function
- [x] Code uses Splashkit functions
- [x] Tested in Chrome and Firefox

omckeon commented:
Update file names to use "example" instead of "simple"

omckeon commented:
> # Peer Review
> I've reviewed the Display X example and apart from the filename issue that @omckeon has addressed, it looks good to me.
> 
> # Checks
> * [x]  All required files are present.
>   
>   * [x]  Title and explanation (.txt)
>   * [x]  C++ code (SplashKit)
>   * [ ]  C++ code (Beyond SplashKit)
>   * [x]  C# code (top-level statements)
>   * [x]  C# code (Object-Oriented Programming)
>   * [x]  Python code
> * [x]  Code correctly uses Splashkit functions.
> * [x]  Code clearly demonstrates the function.
> * [x]  All versions maintain the same structure and comments.
> 
> # Tests done
> * [x]  C++ SplashKit code ran correctly.
> * [ ]  C++ Beyond SplashKit code ran correctly.
> * [x]  C# top level code ran correctly.
> * [x]  C# OOP code ran correctly.
> * [x]  Python code ran correctly.

Note: @SimonRhook and @dijidiji - A few other areas that should be fixed:
- The C++ code file shouldn't have "sk" on the end.
- The example heading in the text file is a bit confusing ("map" has many meanings)
- In the OOP file, the namespace should have "Example" on the end, so that it is not highlighted.
- Local variables use camelCase in C#.
- Remove local variables that are used just 1 time, and then use the function instead (eg. DispCount is not needed)
- Remove any random extra blank lines.
- Check indentation (return 0 line in C++ code).
- Output image should not show which command was used to run the program, just the terminal output section.
- Add blank lines around "using namespace std" when it is included, or just use `std::to_string(...)` instead.
- Just use "int" rather than "uint".

@dijidiji - Hopefully this helps to see the level of detail that reviews/peer-reviews require. It is unlikely that you will approve any PRs on the first review.
Also, please include a screenshot of the example displayed on the site in your local testing as part of your review. (We may need to update the documentation about this.)

cameronhum commented on public/usage-examples/graphics/display_x-2-example.txt:
```diff
@@ -0,0 +1,5 @@
+Draw physical display setup on a window with Display X explaination
```
small spelling mistake. "explaination" => "explanation"

cameronhum reviewed (changes requested):
Hey Simon,

I've reviewed the 2 usage examples for display x. All the examples run successfully and are consistent in function. The website also displays all of them and the note for the resources works as expected. I only had some small issues in regards to the newest example in which some names were not consistent and there were extra spaces. 

The other thing I want to mention which would be great to see for the second usage example is to be a bit more cleaner. As you can see from my screenshot. Due to the large name it goes outside of the rectangle. It would be great if this was resolved to cut of the name with "..." so "Wide viewing angle & High d..." in the case of my screenshot or even word wrapping. I would also like to see a bit more distinction between the title and the displays (Padding below title, etc) and be a bit more descriptive as to what I'm looking at as it took me a while to understand the example since the title doesn't describe whats in the window. However I think the explanation of display x is still important to keep.
Overall I think both usage examples are a good way of showing how to use these in a basic to a more advanced setting in both a graphical and command sense. Good Job! after some small changes I am happy to approve this.
![image](https://github.com/user-attachments/assets/73d3b591-593f-436e-bc4a-543d310c3244)
![image](https://github.com/user-attachments/assets/a5fb13b8-18bf-4553-b46c-b97c558d6aae)
# Peer Review

I've reviewed the Display X - Usage Example

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
  Or, if Beyond SK Pull Request (remove if not relevant):
  - [ ] C++ code (Beyond SplashKit)
- [x] Code correctly uses Splashkit functions.
- [ ] Code clearly demonstrates the function.
- [ ] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [ ] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview

cameronhum commented on public/usage-examples/graphics/display_x-2-example-top-level.cs:
```diff
@@ -0,0 +1,84 @@
+    dispNames[i] = dispName;
+
+    // Set min coordinate offset for drawing
+    if (dispX < minX) minX = dispX;
+    if (dispY < minY) minY = dispY;
```
updated style guide wants if statements not to be one liners

SimonRhook commented:
Hi Cameron,

Thanks for the thorough review, I believe I have addressed all your points except the following: 

> Due to the large name it goes outside of the rectangle. It would be great if this was resolved to cut of the name with "..." so "Wide viewing angle & High d..." in the case of my screenshot or even word wrapping. 

I've considered this but I believe as the focus of this usage example is about showing the x coordinate of a display and not necessarily about the name, the addition of text handling for a long monitor name could introduce unnecessary bloat and complexity in the usage example

Let me know what you think.

cameronhum reviewed (approved):
# Peer Review

I've reviewed the Display X Usage Example and changes have been made and I feel comfortable approving this now. I also agree with not including the suggestion I made due to bloat especially since its already quite a complex example

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
- [x] Code correctly uses Splashkit functions.
- [x] Code clearly demonstrates the function.
- [x] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [x] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview
